include/zephyr/modbus/modbus.h

```diff
@@ -40,6 +40,17 @@ extern "C" {
 #define MODBUS_MBAP_LENGTH		7
 /** Length of MBAP Header plus function code */
 #define MODBUS_MBAP_AND_FC_LENGTH	(MODBUS_MBAP_LENGTH + 1)
+/* Modbus exception codes */
+#define MODBUS_EXC_NONE				0
+#define MODBUS_EXC_ILLEGAL_FC			1
+#define MODBUS_EXC_ILLEGAL_DATA_ADDR		2
+#define MODBUS_EXC_ILLEGAL_DATA_VAL		3
+#define MODBUS_EXC_SERVER_DEVICE_FAILURE	4
+#define MODBUS_EXC_ACK				5
+#define MODBUS_EXC_SERVER_DEVICE_BUSY		6
+#define MODBUS_EXC_MEM_PARITY_ERROR		8
+#define MODBUS_EXC_GW_PATH_UNAVAILABLE		10
+#define MODBUS_EXC_GW_TARGET_FAILED_TO_RESP	11
```

Comment:
Please group these with `@name` + add doxygen docstrings to each define. See below for `@name` usage.

```suggestion
/** @name Modbus exception codes 
 * @{
*/
#define MODBUS_EXC_NONE				0
#define MODBUS_EXC_ILLEGAL_FC			1
#define MODBUS_EXC_ILLEGAL_DATA_ADDR		2
#define MODBUS_EXC_ILLEGAL_DATA_VAL		3
#define MODBUS_EXC_SERVER_DEVICE_FAILURE	4
#define MODBUS_EXC_ACK				5
#define MODBUS_EXC_SERVER_DEVICE_BUSY		6
#define MODBUS_EXC_MEM_PARITY_ERROR		8
#define MODBUS_EXC_GW_PATH_UNAVAILABLE		10
#define MODBUS_EXC_GW_TARGET_FAILED_TO_RESP	11
/** @} */
```

Reply:
@kartben Documentation fixes may also be applied over the next two weeks.

Reply:
Added the section name.